### DISCUSSION TITLE
模版规范 vs 信息输出

### DISCUSSION THREAD

**Discussion by NeverBehave** [DIYgod/RSSHub#8226] [General]
前提：https://github.com/DIYgod/RSSHub/pull/8220 

类似于上述的请求有很多，我们也能理解阅读器对于各种情况处理的方式不是特别好，比如用Telegram订阅的人可能只能看到标题，作者之类的信息出现在标题内会比较方便。**但是，对于项目而言这种情况会导致路由规范度下降，而且每个人对于内容的渲染方式都可能有差异，通用程度差**

我们目前提出一个可能的解决方案：所有路由编写的时候，作者可以声明其中获取的信息，以`JSON`的形式暴露出去。e.g. `/route/feat.json`

在这个里面作者可以暴露全部获取的信息，甚至可以是RSS正常渲染中没有的参数 -- 用途可以是方便调试，或者对于有能力的使用者来说，方便对接第三方服务等

而项目本身则会尽量选择贴近规范的模版进行渲染。同时推行模版规范的前进，方便用户自行替换渲染内容，应该是更长远的道路

_此贴目前仅讨论可能的功能变更，并不是确定的实现方案，欢迎留言补充用例和规则细节_

**Comment by ruleroller** [comment]
模块化很好，方便用户自定义。最好也能推出几种**格式**备选方案，让用户用某种参数的方式，就能够快速实现喜欢的格式输出模块化的结果。

以 #8220 为例，相较于 author：title，更喜欢使用 《title》 by [author] 这样的。 ：）

**Reply by NeverBehave** [reply]
这个的选择主要还是贡献者的做法，view替换可能后续的操作是提供overwrite文件夹在启动时整体替换，参数支持应该是不会考虑的。

而且也说了，RSSHub本身是选择尽量贴合RSS规范的，作者信息应该呆在它该在的地方
